cirq/ops/pauli_sum_exponential.py

```diff
@@ -0,0 +1,131 @@
+# Copyright 2020 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import (Any, Iterator, Tuple, Union, TYPE_CHECKING)
+
+import numpy as np
+import sympy
+
+from cirq import linalg, protocols, value
+from cirq.ops import linear_combinations, pauli_string_phasor
+
+if TYPE_CHECKING:
+    import cirq
+
+
+@value.value_equality(approximate=True)
+class PauliSumExponential:
+    """Represents operator defined by exponential of a PauliSum.
+
+    Given a hermitian/anti-hermitian PauliSum PS_1 + PS_2 + ... + PS_N, this
+    class returns an operation which is equivalent to
+    exp(j * exponent * (PS_1 + PS_2 + ... + PS_N)).
+
+    This class currently supports only exponential of commuting Pauli Terms.
+    """
+
+    def __init__(self,
+                 pauli_sum_like: 'cirq.PauliSumLike',
+                 exponent: Union[int, float, sympy.Basic] = 1,
+                 *,
+                 is_anti_hermitian=False):
```

Comment:
can't we just determine this based on the coefficients? Why does the user need to define this?

Reply:
Modified to determine based on coefficients. The original thought was to specify explicitly so that the action is more clear on the call site but I think the docstring is probably enough to convey the action.